### PR TITLE
add profile page

### PR DESCRIPTION
**Pull request by Segnicho**
Screenshot 2023-04-23 10 46 14 PM
Screenshot 2023-04-23 10 44 42 PM
Screenshot 2023-04-23 10 27 27 PM
I have tried to copy the design. I also used some already existing tailwind components rather than implementing everything by myself as was suggested in my earlier PR.

### CODE REVIEW

File: Web/starter_project_vanguard/public/images/profile-photo.png
Comment (YiseBoge) [inline Web/starter_project_vanguard/public/images/profile-photo.png:1]: let's move this under profile subfolder (i know it will probably be deleted with real data too)

File: Web/starter_project_vanguard/types/user-profile.ts
Comment (YiseBoge) [inline Web/starter_project_vanguard/types/user-profile.ts:1]: [nit] I think `profile` is a fine name for this file, as that is the name we would use in the images and also the components folders

File: Web/starter_project_vanguard/components/profile/AccountSetting.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import React, { useState } from 'react'
+import { AiFillEyeInvisible, AiFillEye } from 'react-icons/ai'
+
+const AccountSetting: React.FC = () => {
+  const [showCurrPassword, setShowCurrPassword] = useState<boolean>(false)
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/components/profile/AccountSetting.tsx:5]: [nit] better name it `showCurrentPassword` instead of `showCurrPassword`
we wouldn't be saving much space

File: Web/starter_project_vanguard/components/profile/AccountSetting.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  const [showNewPassword, setShowNewPassword] = useState<boolean>(false)
+  const [showConfirmPassword, setShowConfirmPassword] = useState<boolean>(false)
+
+  return (
+    <div className="mt-5 flex flex-col justify-center gap-2 px-auto">
+      <div className="flex justify-between py-3">
+        <div className="font-montserrat ">
+          <h1 className="font-semibold text-xl">Manage your account</h1>
+          <p className="text-primary-text">
+            you can change your password here.
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/components/profile/AccountSetting.tsx:15]: [very-nit] let's start with a capital for a sentence

File: Web/starter_project_vanguard/components/profile/AccountSetting.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        </div>
+        <button className="bg-primary text-white rounded px-3 h-9 my-auto mr-8">
+          Save Changes
+        </button>
+      </div>
+      <hr />
+      <form className="mx-auto font-poppins font-semibold text-tertiary-text pt-5 flex flex-col gap-4">
+        <div className="relative">
+          <label htmlFor="currPassword">Current Password</label>
+          <div
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/components/profile/AccountSetting.tsx:26]: [nit] should this be a button if it is clickable?

File: Web/starter_project_vanguard/components/profile/AccountSetting.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+          <p className="text-primary-text">
+            you can change your password here.
+          </p>
+        </div>
+        <button className="bg-primary text-white rounded px-3 h-9 my-auto mr-8">
+          Save Changes
+        </button>
+      </div>
+      <hr />
+      <form className="mx-auto font-poppins font-semibold text-tertiary-text pt-5 flex flex-col gap-4">
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/components/profile/AccountSetting.tsx:23]: I think a flex or a grid might get you better-aligned results
but we can do much UI changes in a different PR

File: Web/starter_project_vanguard/components/profile/PersonalInfo.tsx [hunk 5]
```diff
@@ -0,0 +1,120 @@
+import Image  from 'next/image'
+import React from 'react'
+
+import {User} from '@/types/user-profile'
+import profile from 'public/images/profile-photo.png'
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/components/profile/PersonalInfo.tsx:5]: [nit] i think you should stick to the profilePhoto name for the variable too to stay consistent

File: Web/starter_project_vanguard/components/profile/PersonalInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+const user: User = {
+  firstName: 'Segni',
+  lastName: 'Desta',
+  email: 'segni@gmail.com',
+  img: profile,
+}
```
Comment (Adona12) [inline Web/starter_project_vanguard/components/profile/PersonalInfo.tsx:12]: Even if it's small, I would rather not have mock data here. It would be better to separate it.

File: Web/starter_project_vanguard/components/profile/PersonalInfo.tsx [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+      <div className="flex py-5 gap-10 ">
+        <label htmlFor="" className="mr-10">
+          Your Photo <span className="text-red-500">*</span>
+        </label>
+        <Image
+          src={user.img}
+          alt={user.firstName + user.lastName}
+          className="h-[80px] w-[80px] rounded-lg"
+        />
+        <div className="flex items-center justify-center w-[35%] h-[20%]">
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/components/profile/PersonalInfo.tsx:85]: [nit] if a lot of these input types or labels or important tag are to be reused different places, you can consider making them small components (maybe even inside the same file if it is only used in one file)

File: Web/starter_project_vanguard/public/images/profile-photo.png
Comment (Adona12) [inline Web/starter_project_vanguard/public/images/profile-photo.png:1]: It would be better to put this in a subfolder if you think you're the only one using it. You can have `/profile` subfolder. We're trying to make it consistent everywhere so instead of `/images` subfolder, it should be `/img` subfolder

File: Web/starter_project_vanguard/components/profile/PersonalInfo.tsx [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+              <path d="M2.003 5.884L10 9.882l7.997-3.998A2 2 0 0016 4H4a2 2 0 00-1.997 1.884z"></path>
+              <path d="M18 8.118l-8 4-8-4V14a2 2 0 002 2h12a2 2 0 002-2V8.118z"></path>
+            </svg>
+          </div>
+          <input
+            required
+            value={user.email}
+            type="text"
+            id="input-group-1"
+            className="bg-gray-50 border ml-10 w-[195%] border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block  pl-10 p-2.5  dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
```
Comment (Adona12) [inline Web/starter_project_vanguard/components/profile/PersonalInfo.tsx:71]: It would be better to define the colors globally in the tailwind config file so it can be reused by others.

File: Web/starter_project_vanguard/pages/profile/index.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
+import Link from 'next/link'
+import React, { useState } from 'react'
+
+import AccountSetting from '@/components/profile/AccountSetting'
+import PersonalInfo from '@/components/profile/PersonalInfo'
+
+const Profile: React.FC = () => {
+  const [tab, setTab] = useState<number>(0)
+  const handleTabChange = (tab: number) => {
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/pages/profile/index.tsx:9]: [nit] is there maybe a way to use prebuilt tabs for this? it would make things cleaner if possible but not mandatory for now

File: Web/starter_project_vanguard/pages/profile/index.tsx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+    setTab(tab)
+  }
+
+  return (
+    <div className="px-10">
+      <h1 className="font-medium font-poppins text-[40px]">Profile</h1>
+      <div className="text-sm font-medium text-center text-gray-500 border-b border-gray-200 dark:text-gray-400 dark:border-gray-700">
+        <ul className="flex flex-wrap -mb-px">
+          <li className="mr-2">
+            <Link
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/pages/profile/index.tsx:19]: [nit] does this have to be a link or a button?
or did you want to make it add the navigation to the backstack?

File: Web/starter_project_vanguard/pages/profile/index.tsx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+              } inline-block p-4 rounded-t-lg  dark:text-blue-500 dark:border-blue-500`}
+              aria-current="page"
+            >
+              Account Setting
+            </Link>
+          </li>
+        </ul>
+      </div>
+      {tab === 0 && <PersonalInfo />}
+      {tab === 2 && <AccountSetting />}
```
Comment (YiseBoge) [inline Web/starter_project_vanguard/pages/profile/index.tsx:57]: i think it is better to have a list of these components and render the one based on the tab as the index

**Review by YiseBoge — APPROVED** [review]
overall the code seems good to me (maybe the UI could use some clearing in another PR)
i have left some nits that would be nice to fix before merging tho

File: Web/starter_project_vanguard/pages/profile/index.tsx [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+
+const Profile: React.FC = () => {
+  const [tab, setTab] = useState<number>(0)
+  const handleTabChange = (tab: number) => {
+    setTab(tab)
+  }
+
+  return (
+    <div className="px-10">
+      <h1 className="font-medium font-poppins text-[40px]">Profile</h1>
```
Comment (ruthgdb) [inline Web/starter_project_vanguard/pages/profile/index.tsx:15]: [nit] can you make the font size of this section somewhat proportional to the rest of the page?

File: Web/starter_project_vanguard/components/profile/PersonalInfo.tsx (hunk 6 above)
Comment (Segnicho) [reply Web/starter_project_vanguard/components/profile/PersonalInfo.tsx:12]: In which folder do you think I should put it? (store is for redux, i guess)

**Review by sam5235 — APPROVED** [review]

**Review by sam5235 — APPROVED** [review]
